lib/rubocop/cop/lint/number_convertion.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for usage of #to_i.
+      #
+      # @example
+      #   # bad
+      #   foo.to_i
+      #
+      #   # good
+      #   Integer(foo)
+      class NumberConvertion < Cop
+        MSG = 'Use `Integer()` instead of `#to_i`.'.freeze
+
+        def_node_matcher :to_i?, <<-PATTERN
+          (send (...) :to_i)
```

Comment:
I think the cop should consider `to_i` with a base. For example: `str.to_i(16)` converts the `str` to an integer as hex.